core/src/test/java/org/infinispan/tx/EntryWrappingInterceptorDoesNotBlockTest.java

```diff
@@ -142,11 +143,16 @@ protected void test(int expectRemoteGets, BiFunction<MagicKey, Integer, Object>
       ControlledRpcManager.SentRequest sentPrepare = crm0.expectCommand(PrepareCommand.class).send();
 
       // the node should load all moving keys
+      List<ControlledRpcManager.SentRequest> remoteGets = new ArrayList<>();
       for (int i = 0; i < expectRemoteGets; i++) {
-         crm2.expectCommand(ClusteredGetCommand.class).send().receiveAll();
+         remoteGets.add(crm2.expectCommand(ClusteredGetCommand.class).send());
       }
 
       prepareExpectingInterceptor.await();
+      // we could get an exception if the remote get finished before the prepare finishes
```

Comment:
why?

Reply:
Because the PrepareExpectingInterceptor expects that the invocationstage is receives is not complete. If the remote get is completed before the interceptor gets the stage, the stage can get completed, too, and an assertion fires.

It was not complicated to get the failure when running the test locally, try yourselves on master.

Reply:
@rvansa I couldn't reproduce it locally, that's why I'm asking :)

I would expect the comment to explain why the invocation stage for the prepare command depends on whether the previous remote gets have finished. The JIRA should also have more information, or at least have a stack trace.

Reply:
Ok, I was sending the fix Friday midnight so I didn't bother :) I'll fill description in the JIRA.

Reply:
In my experience the late night fixes need more comments for the others to understand, not the other way around :)

Reply:
@danberindei I've added description to the JIRA, integration is just two clicks away!

Reply:
+1 for the JIRA description, but I still don't think the java comment explains anything... there's no reason why the comments in the code couldn't explain what the test is trying to achieve and how it wants to do it.